lms/djangoapps/ora_staff_grader/views.py

```diff
@@ -206,10 +208,124 @@ def get_assessment_info(self, request, usage_id, submission_uuid):
         """
         Get assessment data from ORA 'get_assessment_info' XBlock.json_handler
         """
-        data = {
-            'submission_uuid': submission_uuid
+        data = {'submission_uuid': submission_uuid}
+        response = call_xblock_json_handler(request, usage_id, 'get_assessment_info', data)
+        return json.loads(response.content)
+
+    def check_submission_lock(self, request, usage_id, submission_uuid):
+        """
+        Look up lock info for the given submission by calling the ORA's 'check_submission_lock' XBlock.json_handler
+        """
+        data = {'submission_uuid': submission_uuid}
+        response = call_xblock_json_handler(request, usage_id, 'check_submission_lock', data)
+        return json.loads(response.content)
+
+
+class UpdateGradeView(RetrieveAPIView):
+    """
+    POST submit a grade for a submission
+
+    Body: {
+        overallFeedback: (string) overall feedback
+        criteria: [
+            {
+                name: (string) name of criterion
+                feedback: (string, optional) feedback for criterion
+                selectedOption: (string) name of selected option or None if feedback-only criterion
+            },
+            ... (one per criteria)
+        ]
+    }
+
+    Response: {
+        gradeStatus: (string) - One of ['graded', 'ungraded']
+        lockStatus: (string) - One of ['unlocked', 'locked', 'in-progress']
+        gradeData: {
+            score: (dict or None) {
+                pointsEarned: (int) earned points
+                pointsPossible: (int) possible points
+            }
+            overallFeedback: (string) overall feedback
+            criteria: (list of dict) [{
+                name: (str) name of criterion
+                feedback: (str) feedback for criterion
+                selectedOption: (str) name of selected option or None if feedback-only criterion
+            }]
         }
-        return call_xblock_json_handler(request, usage_id, 'get_assessment_info', data)
+    }
+    """
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (IsAuthenticated,)
+
+    @require_params([PARAM_ORA_LOCATION, PARAM_SUBMISSION_ID])
+    def post(self, request, ora_location, submission_uuid, *args, **kwargs):
+        # Transform data from frontend format to staff assess format
+        context = {'submission_uuid': submission_uuid}
+        grade_data = StaffAssessSerializer(request.data, context=context).data
+
+        # Perform the staff assessment
+        response = self.submit_grade(request, ora_location, grade_data)
+
+        # Failed response returns 'success': False with an error message
+        if not response['success']:
+            return HttpResponseServerError(response['msg'])
+
+        # TODO - Confirm we want to do unlocking here
```

Comment:
@muselesscreator - confirming, we want to unlock as part of the "submit a grade" step?

Reply:
confirm

Reply:
@muselesscreator - follow-up: assuming this should also re-check that we own the submission lock on the backend before updating grade, yes?